core-dictionary/src/main/java/org/apache/kylin/dict/DictionaryGenerator.java

```diff
@@ -75,13 +75,19 @@ public static IDictionaryBuilder newDictionaryBuilder(DataType dataType) {
         builder.init(dictInfo, baseId, null);
 
         // add values
-        while (valueEnumerator.moveNext()) {
-            String value = valueEnumerator.current();
+        try {
+            while (valueEnumerator.moveNext()) {
+                String value = valueEnumerator.current();
 
-            boolean accept = builder.addValue(value);
+                boolean accept = builder.addValue(value);
 
-            if (accept && samples.size() < nSamples && samples.contains(value) == false)
-                samples.add(value);
+                if (accept && samples.size() < nSamples && samples.contains(value) == false)
+                    samples.add(value);
+            }
+        } catch (Throwable e) {
```

Comment:
Use "Throwable" is not recommended; please use a concrete exception type instead.